stripe/__init__.py

```
@@ -70,3 +62,226 @@ def set_app_info(
+from stripe._stripe_response import StripeResponse as StripeResponse
+from stripe._stripe_response import StripeResponseBase as StripeResponseBase
+from stripe._stripe_response import (
+    StripeStreamResponse as StripeStreamResponse,
+)
+
+# Error types
+from stripe._error import StripeError as StripeError
+from stripe._error import APIError as APIError
+from stripe._error import APIConnectionError as APIConnectionError
+from stripe._error import StripeErrorWithParamCode as StripeErrorWithParamCode
+from stripe._error import CardError as CardError
+from stripe._error import IdempotencyError as IdempotencyError
+from stripe._error import InvalidRequestError as InvalidRequestError
+from stripe._error import AuthenticationError as AuthenticationError
+from stripe._error import PermissionError as PermissionError
+from stripe._error import RateLimitError as RateLimitError
+from stripe._error import (
+    SignatureVerificationError as SignatureVerificationError,
+)
+
+# Backwards compatibility re-exports
+if not TYPE_CHECKING:
+    from stripe import _api_requestor as api_requestor
+    from stripe import _stripe_response as stripe_response
+    from stripe import _stripe_object as stripe_object
+    from stripe import _error_object as error_object
+    from stripe import _error as error
+    from stripe._file import File as FileUpload
+
+    import warnings
+
+    # Python 3.7+ supports module level __getattr__ that allows us to lazy load deprecated modules
+    # this matters because if we pre-load all modules from api_resources while suppressing warning
+    # users will never see those warnings
+    if _sys.version_info[:2] >= (3, 7):
+
+        def __getattr__(name):
+            if name == "abstract":
+                import stripe.api_resources.abstract as _abstract
+
+                return _abstract
+            if name == "api_resources":
+                import stripe.api_resources as _api_resources
+
+                return _api_resources
```

This needs to `raise AttributeError` rather than falling off the end and returning `None`.

- #1158
- #1159